### PR TITLE
chore: restructure dockerfiles

### PR DESCRIPTION
**Pull request by Jonathansumner**
In order to improve maintainability and facilitate the upgrade of the subquery node to v1.19.1, the submodule and local building of `@subql` is removed.
- simplified dockerfiles

---

## Code Review Checklist (to be filled out by reviewer)

- [ ] Description accurately reflects what changes are being made.
- [ ] Either the PR references an issue (via the "Development" combobox) or the description explains the need for the changes.
- [ ] The PR appropriately sized.
- [ ] The PR contains an idempotent DB migration.
- [ ] I have verified the correctness of the DB migration using relevant data (e.g. test-generated data).
- [ ] New code has enough tests.
- [ ] New code has enough documentation to answer "how do I use it?" and "what does it do?".
- [ ] Existing documentation is up-to-date, if impacted.

### CODE REVIEW

File: ci-compose.yml [hunk 1]
```diff
@@ -37,37 +37,35 @@ services:
       DATABASE_URL: "postgres://subquery:subquery@postgres/subquery"
       SHADOW_DATABASE_URL: "postgres://subquery:subquery@postgres/subquery_shadow"
       ROOT_DATABASE_URL: "postgres://subquery:subquery@postgres/postgres"
+    volumes:
+      - ./:/app
     command:
       - -f=/app
       - --db-schema=app
       - --batch-size=1
+      - --disable-historical=true
```
[the remark this answers is not in the document]
Comment (Jonathansumner) [reply ci-compose.yml:46]: https://academy.subquery.network/run_publish/historical.html

Interesting feature that I thought I'd include support for, however, it adds some extra fields that would require a revamp of the test suite to include some default values

[the remark this answers is not in the document]
Comment (Jonathansumner) [reply ci-compose.yml:46]: https://academy.subquery.network/run_publish/historical.html

Interesting feature that I thought I'd include support for, however, it adds some extra fields that would require a revamp of the test suite to include some default values

File: ci-compose.yml [hunk 2]
```diff
@@ -37,37 +37,35 @@ services:
...
       test: ["CMD", "curl", "-f", "http://subquery-node:3000/ready"]
       interval: 3s
       timeout: 5s
       retries: 10
 
   graphql-engine:
-    build:
-      context: .
-      dockerfile: ./docker/api.dockerfile
+    image: onfinality/subql-query:latest
```
[the remark this answers is not in the document]
Comment (Jonathansumner) [reply ci-compose.yml:54]: I decided to reintroduce this, as a slight delay does actually help stability on startup

[the remark this answers is not in the document]
Comment (Jonathansumner) [reply ci-compose.yml:54]: I decided to reintroduce this, as a slight delay does actually help stability on startup

File: tests/e2e/entities/test_native_primitives.py [hunk 3]
```diff
@@ -359,13 +351,9 @@ def filtered_messages_query(_filter, order=""):
                 order_messages_by_block_height_asc: self.assertGreaterEqual,
                 order_messages_by_block_height_desc: self.assertLessEqual,
             },
-            "events": {
-                order_events_by_block_height_asc: self.assertGreaterEqual,
-                order_events_by_block_height_desc: self.assertLessEqual,
-            },
         }
 
-        for key in ["transactions", "messages", "events"]:
```
Comment (MissingNO57) [inline tests/e2e/entities/test_native_primitives.py:368]: Why is this changed?

**Review by MissingNO57 — APPROVED** [review]
Great job!
